src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -232,114 +229,92 @@ internal static bool HasIllegalCharacters(string path, bool checkAdditional = fa
         /// <summary>
         /// Only check for ? and *.
         /// </summary>
-        internal static bool HasAdditionalIllegalCharacters(string path)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal unsafe static bool HasAdditionalIllegalCharacters(string path)
         {
             int startIndex = PathInternal.IsExtended(path) ? ExtendedPathPrefix.Length : 0;
-            return path.IndexOfAny(AdditionalInvalidPathChars, startIndex) >= 0;
+
+            char currentChar;
+            for (int i = startIndex; i < path.Length; i++)
+            {
+                currentChar = path[i];
+                if (currentChar == '*' || currentChar == '?') return true;
+            }
+            return false;
```

Comment:
Not the same. His was an optimized path for a single char. This was measurably faster for the two char case- the cycles spent went way down.